javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -333,11 +333,20 @@ public static CharSeq repeat(char character, int times) {
      * @return A CharSeq representing {@code this * times}
      */
     public CharSeq repeat(int times) {
-        final StringBuilder builder = new StringBuilder();
-        for (int i = 0; i < times; i++) {
-            builder.append(back);
+        if (times <= 0) return empty();
+        else if (times == 1) return this;
+        else {
+            final int finalLength = length() * times;
+            final StringBuilder result = new StringBuilder(2 * finalLength);
+            result.append(back);
+
+            for (int i = 1; i <= times; i *= 2) {
+                result.append(result);
+            }
+            result.setLength(finalLength);
```

Comment:
Hmm, it seems to me their version is buggy for cases when the repetition count is a power of 2.
Will provide a PR for Guava also (luckly I didn't copy their impl, so I could compare it against mine).